### DISCUSSION TITLE
Allow StandardServiceRegistryBuilder.configure() if the hibernate.cfg.xml is not found using the prior ways to set it up

### DISCUSSION THREAD

**Discussion by chhil** [jpos/jPOS-EE#252] [Q&A] [answered]
https://github.com/jpos/jPOS-EE/blob/1f857edcd47f1330daf14c4efb9e530ddf7bfcda/modules/dbsupport/src/main/java/org/jpos/ee/DB.java#L554

Does it make sense to do the following instead?

```
                if (hibCfg == null) {
                    hibCfg = System.getProperty("HIBERNATE_CFG", "/hibernate.cfg.xml");
                    if (getClass().getClassLoader()
                                  .getResource(hibCfg) != null) {
                        ssrb.configure(hibCfg);
                    }
                    else {
                        ssrb.configure();
                    }
                }
```
Let it find the default hibernate.cfg.xml that can be anywhere on the classpath.

My hibernate.cfg.xml resides in the cfg folder and cfg folder is on my classpath.

**Comment by ar** [comment, marked as answer]
jPOS-EE expects the template file `hibernate.cfg.xml` at the top of the classpath. In order to override the default with a custom `hibernate.cfg.xml` one needs to call `DB db = new DB ("path/to/hibernate.cfg.xml")` or configure `HIBERNATE_CFG` property (that can be done from code using `System.setProperty("HIBERNATE_CFG", "path/to/hibernate.cfg.xml")` once from your application.

That said, we'll think about an easy way to override it using a convention name (that could be `cfg/hibernate.cfg.xml`)

I'm thinking along the lines of:

- If there's a `cfg/hibernate.cfg.xml` in the filesystem and HIBERNATE_CFG environment variable is not present, we use that file
- otherwise, we keep the existing behavior

Would that work for you @chhil  ?

**Reply by chhil** [reply]
I am currently leaning towards using
```
System.setProperty("HIBERNATE_CFG", "path/to/hibernate.cfg.xml")
```

We have a qbean that sets various system and security properties, will add the HIBERNATE_CFG to it.
